### ISSUE TITLE
Highlighting for strings with "$(...)" is inconsistent 

### ISSUE THREAD

**Issue by vors** [PowerShell/EditorSyntax#106]
Moved from https://github.com/SublimeText/PowerShell/issues/66

### Environment

- Editor and Version (VS Code, Atom, Sublime): Sublime
- Your primary theme: Default

### Issue Description

`$(...)` inside a string is an expression interpolation.
Ideally it should be highlighted exectly the same way as it's highlighted when put outside of the string context.
I'm not sure it's possible to achive with regex grammar.

#### Screenshots

Sublime -- difference between interpolated and non-interpolated
![image](https://cloud.githubusercontent.com/assets/816680/5788170/4624ffda-9ddc-11e4-9ad5-cf36b4f57534.png)

### Expected Behavior

PowerShell_ISE -- both are the same
![image](https://cloud.githubusercontent.com/assets/816680/5788164/221d23c4-9ddc-11e4-9fee-187e412d092d.png)

### Code Samples

```PowerShell
"blablabla $(invoke-foo baz $a.bar) blablabla"
invoke-foo baz $a.bar
```

**Comment by vors** [comment]
Yeah, sublime-syntax is more powerful then tmLanguage, but the benefits of having a common EditorSyntax (that we finally got) outweigh the benefits of more expressive grammar rules.

**Comment by msftrncs** [comment]
also see the images in this issue:

https://github.com/PowerShell/vscode-powershell/issues/1401

It shows the static methods referenced in the first couple lines are colored wrong.  Didn't realize it right away, as everything else looks pretty consistent.
